openmls/src/group/mls_group/membership.rs

```diff
@@ -197,6 +199,7 @@ impl MlsGroup {
                 remove_proposal.clone(),
             )?);
 
+        self.reset_aad();
```

Comment:
This looks like a lot of `reset_aad` calls. How do you ensure that it's always reset when needed?

Reply:
The test covers some of them, but not all of them are. If you think that's necessary I can file an issue for a follow-up.

Reply:
I think 100% test coverage here is difficult. I'm wondering more if we could put the reset somewhere where we can be certain that it's always called. I'm fine getting this in as is, but maybe file an issue to ensure that we always reset (if that's with tests or a different style).

Reply:
https://github.com/openmls/openmls/issues/1622